zio-http/src/main/scala/zhttp/service/client/ClientChannelInitializer.scala

```diff
@@ -1,14 +1,23 @@
 package zhttp.service.client
 
+import io.netty.channel.ChannelPipeline
 import io.netty.handler.codec.http.{HttpClientCodec => JHttpClientCodec}
+import io.netty.handler.ssl.SslContext
 import zhttp.core.{JChannel, JChannelHandler, JChannelInitializer, JHttpObjectAggregator}
 
-final case class ClientChannelInitializer[R](channelHandler: JChannelHandler) extends JChannelInitializer[JChannel]() {
+final case class ClientChannelInitializer[R](channelHandler: JChannelHandler, port: Int)
+    extends JChannelInitializer[JChannel]() {
   override def initChannel(ch: JChannel): Unit = {
-    ch.pipeline()
+    val sslCtx: Option[SslContext] = if (port == 443) Some(ClientSSLContext.getSSLContext()) else None
```

Comment:
- `port = 443` will not guarantee the request to be HTTPS
- Unnecessary creation of `Option` is going to be expensive.